tools/pipelines/templates/include-publish-npm-package-steps.yml

```diff
@@ -53,114 +53,20 @@ steps:
-                  then
-                    echo Package has already been published.
-                    local_f="${f}_local"
-                    mv "$f" "$local_f"
-                    package_name=$(grep -oP 'npm notice package: \K.*' "publish_log")
-                    npm pack $package_name
-                    if cmp -s "$f" "$local_f"
-                    then
-                      echo Continuing as published package matches the current one that was attempting to be released.
-                      rm "$f"
-                      mv "$local_f" "$f"
-                      break
-                    else
-                      echo ERROR: Published package does not match the current one attempting to be released for the same version.
-                    fi
-                  fi
-                  rm "$f"
-                  mv "$local_f" "$f"
-                  exit 1
-                fi
-            else
-              feedName=$(grep -oP '_packaging/\K[^/]+(?=/npm/)' publish_log)
-              echo "Feed Name: $feedName"
 
-              if [[ "$(release)" == "release" && $feedName == "build" ]]; then
-                packageVersion=$(grep -oP 'npm notice version:\s+\K[^\s]+' publish_log)
-
-                if [ "${{ parameters.artifactPath }}" == "scoped" ]; then
-                  scopes=("fluidframework" "fluid-tools" "fluid-internal" "fluid-experimental" "fluid-private" "fluid-example")
-                  for scope in "${scopes[@]}"; do
-                      if [[ "$packageName" == "$scope"* ]]; then
-                          packageName="@${packageName/$scope-/$scope/}"
-                      fi
-                  done
-                fi
-
-                feedPromotionUrl="$(ado-feeds-api-base)/packaging/feeds/${feedName}/npm/${packageName}/versions/${packageVersion}?api-version=7.1-preview.1"
+      cp .npmrc ~/.npmrc
 
-                echo "Promoting ${packageName} to release view using: ${feedPromotionUrl}"
-                response=$(curl -f -s -X PATCH \
-                  -H "Content-Type: application/json" \
-                  -H "Accept: application/json" \
-                  -u ":$(System.AccessToken)" \
-                  -d '{
-                        "views": {
-                          "op": "add",
-                          "path": "/views/-",
-                          "value": "Release"
-                        }
-                      }' \
-                      "$feedPromotionUrl"
-                  )
+      maximumRetryIfNetworkError=3
+      tarballsDir=$(Pipeline.Workspace)/pack/${{ parameters.artifactPath }}
+      orderFile=$(Pipeline.Workspace)/pack/packagePublishOrder-${{ parameters.feedKind }}.txt
 
-                curlExitStatus=$?
+      # echo the command for debugging
+      echo "flub publish tarballs --dir $tarballsDir --orderFile $orderFile --publishArgs \"$tag\" --retry $maximumRetryIfNetworkError --verbose"
```

Comment:
Is there a way in bash to execute a string? I always expect that lines will end up diverging.
Or a way to turn on command echoing?
I feel like I saw magic in some pipeline where task had the script content shown - but I don't remember where I saw that and I don't see any doc about it.
Otherwise, I'd drop the `# echo` comment and move the `# execute` comment above them both. Easier on the eye to diff.

Reply:
If I recall correctly [this](https://github.com/microsoft/FluidFramework/blob/b599bc7d0b179d1ec12289bca630d661b17380c4/scripts/pack-packages.sh#L2) turns on command echoing, among other things we should probably do here as well.

Reply:
It's the `set -x`. I added it in latest and removed the echo. I also reformatted the command to be more readable.